src/Socket/StreamFunctions.php

```diff
@@ -68,11 +68,13 @@ public function fread($handle, $length)
         return fread($handle, $length);
     }
 
-    public function fsockopen($hostname, $port = -1, &$errno = null, &$errstr = null, $timeout = null)
+    public function fsockopen($hostname, $port = -1, &$errno = null, &$errstr = null, $timeout = null, $persistent = false)
     {
-        // Warnings (e.g. connection refused) suppressed;
-        // return value, $errno and $errstr should be checked instead.
-        return @fsockopen($hostname, $port, $errno, $errstr, $timeout);
+        if ($persistent) {
+            return @pfsockopen($hostname, $port, $errno, $errstr, $timeout);
+        } else {
+            return @fsockopen($hostname, $port, $errno, $errstr, $timeout);
+        }
```

Comment:
This conditional should be moved up to `NativeSocket::construct()`, and split into two separate methods here. `StreamFunctions` methods should map simply to their underlying PHP functions.